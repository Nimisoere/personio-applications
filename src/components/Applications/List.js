import React from "react";
import PropTypes from "prop-types";
import DataTable from "../Shared/DataTable";
import { MdLens } from "react-icons/md";
import { appConstants } from "../../_constants";
const calculateAge = dob => {
  const today = new Date();
  const birthDate = new Date(dob);
  let age = today.getFullYear() - birthDate.getFullYear();
  var m = today.getMonth() - birthDate.getMonth();
  if (m < 0 || (m === 0 && today.getDate() < birthDate.getDate())) {
    age--;
  }
  return age;
};

export const ApplicationsList = ({ ...props }) => {
  const { fetching, data, loadData, error } = props;
  const dataCount =
    data && data.count ? data.count : data && data.data ? data.data.length : 0;

  const columns = [
    {
      name: "User ID",
      accessor: "id"
    },
    {
      name: "Name",
      accessor: "name",
      filterable: true
    },
    {
      name: "Email",
      accessor: "email"
    },
    {
      name: "Date of Birth / Age",
      accessor: "birth_date",
      Cell: row => {
        return (
          <div>
            {row.birth_date} <strong>({calculateAge(row.birth_date)})</strong>
          </div>
        );
      }
    },
    {
      name: "Years of experience",
      accessor: "year_of_experience",
      sortable: true
    },
    {
      name: "Position Applied",
      accessor: "position_applied",
      sortable: true,
      filterable: true
    },
    {
      name: "Application Date",
      accessor: "application_date",
      sortable: true,
      sortType: "date"
    },
    {
      name: "Application Status",
      accessor: "status",
      filterable: true,
      filterOptions: [
        {value: appConstants.APPLICATION_STATUS.APPROVED, label: "Approved"},
        {value: appConstants.APPLICATION_STATUS.WAITING, label: "Waiting"},
        {value: appConstants.APPLICATION_STATUS.REJECTED, label: "Rejected"}
      ],
      Cell: row => {
        switch (row.status) {
          case appConstants.APPLICATION_STATUS.APPROVED:
            return (
              <span>
                <MdLens className="text-success" /> Approved
              </span>
            );
          case appConstants.APPLICATION_STATUS.REJECTED:
            return (
              <span>
                <MdLens className="text-danger" /> Rejected
              </span>
            );
          case appConstants.APPLICATION_STATUS.WAITING:
            return (
              <span>
                <MdLens className="text-warning" /> Waiting
              </span>
            );
          default:
            return "foo";
        }
      }
    }
  ];

  return (
    <div>
      <DataTable
        columns={columns}
        loading={fetching}
        data={data}
        count={dataCount}
        countName="Applications"
        defaultPageSize="10"
        loadData={loadData}
        error={error}
        bordered={true}
        striped={true}
        hover={true}
        responsive
      />
    </div>
  );
};

ApplicationsList.propTypes = {
  match: PropTypes.object.isRequired,
  data: PropTypes.object,
  loadData: PropTypes.func,
  fetching: PropTypes.bool,
  fetched: PropTypes.bool,
  error: PropTypes.object
};

export default ApplicationsList;
