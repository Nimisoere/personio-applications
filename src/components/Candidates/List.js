import React from "react";
import { Row, Col, Button, Alert } from "reactstrap";
import { MdAutorenew } from "react-icons/md";

import PropTypes from "prop-types";

import { Spinner } from "../Shared";
import { message } from "../../_constants";
import DataTable from "../Shared/DataTable";

export const CandidatesList = ({ ...props }) => {
  const { fetching, data, loadData, error } = props;

  return (
    <div>
      <Row>
        <Col sm="6">
          <h5 className="item-count">{data && data.count} Result(s) Found</h5>
        </Col>
        <Col sm="6">
          <div className="text-right">
            <Button
              disabled={fetching}
              onClick={loadData}
              className="btn btn-primary btn-outline"
            >
              {fetching ? (
                <Spinner />
              ) : (
                <span>
                  <MdAutorenew size={20} /> Refresh
                </span>
              )}
            </Button>
          </div>
        </Col>
      </Row>
      <hr />
      {error && <Alert color="danger">{message.OUTDATED_DATA}</Alert>}
      <DataTable />
    </div>
  );
};

CandidatesList.propTypes = {
  match: PropTypes.object.isRequired,
  data: PropTypes.object,
  loadData: PropTypes.func,
  fetching: PropTypes.bool,
  fetched: PropTypes.bool,
  error: PropTypes.object
};

export default CandidatesList;
