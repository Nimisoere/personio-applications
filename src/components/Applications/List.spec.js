import React from "react";
import Enzyme, { shallow } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import { ApplicationsList } from "./List";

Enzyme.configure({ adapter: new Adapter() });

function setup() {
  const props = {
    match: "",
    fetching: false,
    data: {},
    error: {},
    loadData: jest.fn()
  };
  const enzymeWrapper = shallow(<ApplicationsList {...props} />);
  return {
    props,
    enzymeWrapper
  };
}

describe("components", () => {
  describe("Applications List", () => {
    it("should render", () => {
      const { enzymeWrapper } = setup();
      expect(enzymeWrapper).toMatchSnapshot();
    });
  });
});
