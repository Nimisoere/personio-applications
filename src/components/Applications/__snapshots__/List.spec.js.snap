// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components Applications List should render 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ApplicationsList
    data={Object {}}
    error={Object {}}
    fetching={false}
    loadData={[MockFunction]}
    match=""
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Row
          tag="div"
        >
          <Col
            sm="6"
            tag="div"
            widths={
              Array [
                "xs",
                "sm",
                "md",
                "lg",
                "xl",
              ]
            }
          >
            <h5
              className="item-count"
            >
               Result(s) Found
            </h5>
          </Col>
          <Col
            sm="6"
            tag="div"
            widths={
              Array [
                "xs",
                "sm",
                "md",
                "lg",
                "xl",
              ]
            }
          >
            <div
              className="text-right"
            >
              <Button
                className="btn btn-primary btn-outline"
                color="secondary"
                disabled={false}
                onClick={[MockFunction]}
                tag="button"
              >
                <span>
                  <MdAutorenew
                    size={20}
                  />
                   Refresh
                </span>
              </Button>
            </div>
          </Col>
        </Row>,
        <hr />,
        <Alert
          closeAriaLabel="Close"
          color="danger"
          fade={true}
          isOpen={true}
          tag="div"
          transition={
            Object {
              "appear": true,
              "baseClass": "fade",
              "baseClassActive": "show",
              "enter": true,
              "exit": true,
              "in": true,
              "mountOnEnter": false,
              "onEnter": [Function],
              "onEntered": [Function],
              "onEntering": [Function],
              "onExit": [Function],
              "onExited": [Function],
              "onExiting": [Function],
              "tag": "div",
              "timeout": 150,
              "unmountOnExit": true,
            }
          }
        >
          The data displayed below might be outdated. Click 'Refresh' to update.
        </Alert>,
        <DataTable />,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": Array [
            <Col
              sm="6"
              tag="div"
              widths={
                Array [
                  "xs",
                  "sm",
                  "md",
                  "lg",
                  "xl",
                ]
              }
            >
              <h5
                className="item-count"
              >
                 Result(s) Found
              </h5>
            </Col>,
            <Col
              sm="6"
              tag="div"
              widths={
                Array [
                  "xs",
                  "sm",
                  "md",
                  "lg",
                  "xl",
                ]
              }
            >
              <div
                className="text-right"
              >
                <Button
                  className="btn btn-primary btn-outline"
                  color="secondary"
                  disabled={false}
                  onClick={[MockFunction]}
                  tag="button"
                >
                  <span>
                    <MdAutorenew
                      size={20}
                    />
                     Refresh
                  </span>
                </Button>
              </div>
            </Col>,
          ],
          "tag": "div",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": <h5
                className="item-count"
              >
                 Result(s) Found
              </h5>,
              "sm": "6",
              "tag": "div",
              "widths": Array [
                "xs",
                "sm",
                "md",
                "lg",
                "xl",
              ],
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  undefined,
                  " Result(s) Found",
                ],
                "className": "item-count",
              },
              "ref": null,
              "rendered": Array [
                undefined,
                " Result(s) Found",
              ],
              "type": "h5",
            },
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": <div
                className="text-right"
              >
                <Button
                  className="btn btn-primary btn-outline"
                  color="secondary"
                  disabled={false}
                  onClick={[MockFunction]}
                  tag="button"
                >
                  <span>
                    <MdAutorenew
                      size={20}
                    />
                     Refresh
                  </span>
                </Button>
              </div>,
              "sm": "6",
              "tag": "div",
              "widths": Array [
                "xs",
                "sm",
                "md",
                "lg",
                "xl",
              ],
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <Button
                  className="btn btn-primary btn-outline"
                  color="secondary"
                  disabled={false}
                  onClick={[MockFunction]}
                  tag="button"
                >
                  <span>
                    <MdAutorenew
                      size={20}
                    />
                     Refresh
                  </span>
                </Button>,
                "className": "text-right",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": <span>
                    <MdAutorenew
                      size={20}
                    />
                     Refresh
                  </span>,
                  "className": "btn btn-primary btn-outline",
                  "color": "secondary",
                  "disabled": false,
                  "onClick": [MockFunction],
                  "tag": "button",
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      <MdAutorenew
                        size={20}
                      />,
                      " Refresh",
                    ],
                  },
                  "ref": null,
                  "rendered": Array [
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "function",
                      "props": Object {
                        "size": 20,
                      },
                      "ref": null,
                      "rendered": null,
                      "type": [Function],
                    },
                    " Refresh",
                  ],
                  "type": "span",
                },
                "type": [Function],
              },
              "type": "div",
            },
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": "hr",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": "The data displayed below might be outdated. Click 'Refresh' to update.",
          "closeAriaLabel": "Close",
          "color": "danger",
          "fade": true,
          "isOpen": true,
          "tag": "div",
          "transition": Object {
            "appear": true,
            "baseClass": "fade",
            "baseClassActive": "show",
            "enter": true,
            "exit": true,
            "in": true,
            "mountOnEnter": false,
            "onEnter": [Function],
            "onEntered": [Function],
            "onEntering": [Function],
            "onExit": [Function],
            "onExited": [Function],
            "onExiting": [Function],
            "tag": "div",
            "timeout": 150,
            "unmountOnExit": true,
          },
        },
        "ref": null,
        "rendered": "The data displayed below might be outdated. Click 'Refresh' to update.",
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Row
            tag="div"
          >
            <Col
              sm="6"
              tag="div"
              widths={
                Array [
                  "xs",
                  "sm",
                  "md",
                  "lg",
                  "xl",
                ]
              }
            >
              <h5
                className="item-count"
              >
                 Result(s) Found
              </h5>
            </Col>
            <Col
              sm="6"
              tag="div"
              widths={
                Array [
                  "xs",
                  "sm",
                  "md",
                  "lg",
                  "xl",
                ]
              }
            >
              <div
                className="text-right"
              >
                <Button
                  className="btn btn-primary btn-outline"
                  color="secondary"
                  disabled={false}
                  onClick={[MockFunction]}
                  tag="button"
                >
                  <span>
                    <MdAutorenew
                      size={20}
                    />
                     Refresh
                  </span>
                </Button>
              </div>
            </Col>
          </Row>,
          <hr />,
          <Alert
            closeAriaLabel="Close"
            color="danger"
            fade={true}
            isOpen={true}
            tag="div"
            transition={
              Object {
                "appear": true,
                "baseClass": "fade",
                "baseClassActive": "show",
                "enter": true,
                "exit": true,
                "in": true,
                "mountOnEnter": false,
                "onEnter": [Function],
                "onEntered": [Function],
                "onEntering": [Function],
                "onExit": [Function],
                "onExited": [Function],
                "onExiting": [Function],
                "tag": "div",
                "timeout": 150,
                "unmountOnExit": true,
              }
            }
          >
            The data displayed below might be outdated. Click 'Refresh' to update.
          </Alert>,
          <DataTable />,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": Array [
              <Col
                sm="6"
                tag="div"
                widths={
                  Array [
                    "xs",
                    "sm",
                    "md",
                    "lg",
                    "xl",
                  ]
                }
              >
                <h5
                  className="item-count"
                >
                   Result(s) Found
                </h5>
              </Col>,
              <Col
                sm="6"
                tag="div"
                widths={
                  Array [
                    "xs",
                    "sm",
                    "md",
                    "lg",
                    "xl",
                  ]
                }
              >
                <div
                  className="text-right"
                >
                  <Button
                    className="btn btn-primary btn-outline"
                    color="secondary"
                    disabled={false}
                    onClick={[MockFunction]}
                    tag="button"
                  >
                    <span>
                      <MdAutorenew
                        size={20}
                      />
                       Refresh
                    </span>
                  </Button>
                </div>
              </Col>,
            ],
            "tag": "div",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "children": <h5
                  className="item-count"
                >
                   Result(s) Found
                </h5>,
                "sm": "6",
                "tag": "div",
                "widths": Array [
                  "xs",
                  "sm",
                  "md",
                  "lg",
                  "xl",
                ],
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    undefined,
                    " Result(s) Found",
                  ],
                  "className": "item-count",
                },
                "ref": null,
                "rendered": Array [
                  undefined,
                  " Result(s) Found",
                ],
                "type": "h5",
              },
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "children": <div
                  className="text-right"
                >
                  <Button
                    className="btn btn-primary btn-outline"
                    color="secondary"
                    disabled={false}
                    onClick={[MockFunction]}
                    tag="button"
                  >
                    <span>
                      <MdAutorenew
                        size={20}
                      />
                       Refresh
                    </span>
                  </Button>
                </div>,
                "sm": "6",
                "tag": "div",
                "widths": Array [
                  "xs",
                  "sm",
                  "md",
                  "lg",
                  "xl",
                ],
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <Button
                    className="btn btn-primary btn-outline"
                    color="secondary"
                    disabled={false}
                    onClick={[MockFunction]}
                    tag="button"
                  >
                    <span>
                      <MdAutorenew
                        size={20}
                      />
                       Refresh
                    </span>
                  </Button>,
                  "className": "text-right",
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "children": <span>
                      <MdAutorenew
                        size={20}
                      />
                       Refresh
                    </span>,
                    "className": "btn btn-primary btn-outline",
                    "color": "secondary",
                    "disabled": false,
                    "onClick": [MockFunction],
                    "tag": "button",
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": Array [
                        <MdAutorenew
                          size={20}
                        />,
                        " Refresh",
                      ],
                    },
                    "ref": null,
                    "rendered": Array [
                      Object {
                        "instance": null,
                        "key": undefined,
                        "nodeType": "function",
                        "props": Object {
                          "size": 20,
                        },
                        "ref": null,
                        "rendered": null,
                        "type": [Function],
                      },
                      " Refresh",
                    ],
                    "type": "span",
                  },
                  "type": [Function],
                },
                "type": "div",
              },
              "type": [Function],
            },
          ],
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": "hr",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": "The data displayed below might be outdated. Click 'Refresh' to update.",
            "closeAriaLabel": "Close",
            "color": "danger",
            "fade": true,
            "isOpen": true,
            "tag": "div",
            "transition": Object {
              "appear": true,
              "baseClass": "fade",
              "baseClassActive": "show",
              "enter": true,
              "exit": true,
              "in": true,
              "mountOnEnter": false,
              "onEnter": [Function],
              "onEntered": [Function],
              "onEntering": [Function],
              "onExit": [Function],
              "onExited": [Function],
              "onExiting": [Function],
              "tag": "div",
              "timeout": 150,
              "unmountOnExit": true,
            },
          },
          "ref": null,
          "rendered": "The data displayed below might be outdated. Click 'Refresh' to update.",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
